### PR TITLE
Best buy handler

### PR DESCRIPTION
**Pull request by JeyachandranV**

### CODE REVIEW

File: superagentx_handlers/best_buy.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+from abc import ABC
+import aiohttp
+from superagentx.handler.base import BaseHandler
+from superagentx.handler.decorators import tool
+
+base_url = "https://api.bestbuy.com/v1/products"
```
Comment (SyedHabibM) [inline superagentx_handlers/best_buy.py:6]: Make this as `constant` variable, (ie). Upper Case

File: superagentx_handlers/best_buy.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+show_options = ("show=customerReviewAverage"
```
Comment (SyedHabibM) [inline superagentx_handlers/best_buy.py:7]: Make this as `constant` variable, (ie). Upper Case. And align this as per `pep8`

File: superagentx_handlers/best_buy.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+                ",modelNumber"
+                ",name"
+                ",onlineAvailability"
+                ",onSale"
+                ",percentSavings"
+                ",regularPrice"
+                ",salePrice"
+                ",sku"
+                ",thumbnailImage")
+default_pagination = "&pageSize=100"
```
Comment (SyedHabibM) [inline superagentx_handlers/best_buy.py:21]: Make this as `constant` variable, (ie). Upper Case

File: superagentx_handlers/best_buy.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+response_format = "&format=json"
```
Comment (SyedHabibM) [inline superagentx_handlers/best_buy.py:22]: Make this as `constant` variable, (ie). Upper Case

File: superagentx_handlers/best_buy.py [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+
+class BestbuyHandler(BaseHandler, ABC):
```
Comment (SyedHabibM) [inline superagentx_handlers/best_buy.py:25]: Remove `ABC`

File: superagentx_handlers/best_buy.py [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+        Attributes:
+            api_key (str): The API key for authenticating requests to the Best Buy API.
+
+        Methods:
+            get_best_buy_info(keyword: str, pagination: str = None) -> dict:
+                Retrieves product information from the Best Buy API based on the specified keyword.
+    """
+
+    def __init__(self,
```
Comment (SyedHabibM) [inline superagentx_handlers/best_buy.py:38]: Align this

File: superagentx_handlers/best_buy.py [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+                ",salePrice"
+                ",sku"
+                ",thumbnailImage")
+default_pagination = "&pageSize=100"
+response_format = "&format=json"
+
+
+class BestbuyHandler(BaseHandler, ABC):
+    """
+        A handler for interacting with the Best Buy API. This class provides methods
```
Comment (SyedHabibM) [inline superagentx_handlers/best_buy.py:27]: Align this doc str

File: superagentx_handlers/best_buy.py [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+    def __init__(self,
+                 *,
+                 api_key: str
+                 ):
+        super().__init__()
+        self.api_key = api_key
+
+    @tool
+    async def get_best_buy_info(self, keyword: str, pagination: str = None):
+        """
```
Comment (SyedHabibM) [inline superagentx_handlers/best_buy.py:47]: Align this doc str

File: superagentx_handlers/best_buy.py [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+
+    def __init__(self,
+                 *,
+                 api_key: str
+                 ):
+        super().__init__()
+        self.api_key = api_key
+
+    @tool
+    async def get_best_buy_info(self, keyword: str, pagination: str = None):
```
Comment (SyedHabibM) [inline superagentx_handlers/best_buy.py:46]: Align the args. And rename `keyword` argument to some other valid name

File: tests/handlers/test_best_buy.py [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+ Run Pytest:  
+
+   1. pytest --log-cli-level=INFO tests/handlers/test_best_buy.py::TestBestBuy::test_get_best_buy_info
+
+'''
+
+
+@pytest.fixture
+async def best_buy_handler() -> BestbuyHandler:
+    return BestbuyHandler(api_key="dmx8sagAYsaH0ghrgAA7OhkR")
```
Comment (SyedHabibM) [inline tests/handlers/test_best_buy.py:16]: Remove this api key

File: superagentx_handlers/best_buy.py (hunk 1 above)
Comment (JeyachandranV) [reply superagentx_handlers/best_buy.py]: changed to uppercase

File: superagentx_handlers/best_buy.py (hunk 2 above)
Comment (JeyachandranV) [reply superagentx_handlers/best_buy.py]: changed to uppercase, aligned as per pep8 rule

File: superagentx_handlers/best_buy.py (hunk 3 above)
Comment (JeyachandranV) [reply superagentx_handlers/best_buy.py]: changed to uppercase

File: superagentx_handlers/best_buy.py (hunk 4 above)
Comment (JeyachandranV) [reply superagentx_handlers/best_buy.py]: changed to uppercase

File: superagentx_handlers/best_buy.py (hunk 5 above)
Comment (JeyachandranV) [reply superagentx_handlers/best_buy.py]: Removed

File: superagentx_handlers/best_buy.py (hunk 7 above)
Comment (JeyachandranV) [reply superagentx_handlers/best_buy.py]: Aligned

File: superagentx_handlers/best_buy.py (hunk 6 above)
Comment (JeyachandranV) [reply superagentx_handlers/best_buy.py:43]: Aligned

File: superagentx_handlers/best_buy.py (hunk 9 above)
Comment (JeyachandranV) [reply superagentx_handlers/best_buy.py]: Aligned and name changed instead keyword to "search_text"

File: superagentx_handlers/best_buy.py (hunk 8 above)
Comment (JeyachandranV) [reply superagentx_handlers/best_buy.py:52]: Aligned

File: tests/handlers/test_best_buy.py (hunk 10 above)
Comment (JeyachandranV) [reply tests/handlers/test_best_buy.py]: Removed

**Review by SyedHabibM — APPROVED** [review]

**Pull request closed without merging**